www/_template/reference/configuration.md

```diff
@@ -5,110 +5,325 @@ description: The Snowpack configuration API reference.
+{
+  plugins: ['plugin-1', ['plugin-2', { 'plugin-option': false }]];
+}
+```
+
+## config.devOptions
+
+**Type**: `object` (option name: value)
+
+Configure the Snowpack dev server.
+
+### devOptions.secure
+
+**Type**: `boolean`
+
+**Default**: `false`
+
+Toggles whether Snowpack dev server should use HTTPS with HTTP2 enabled.
+
+### devOptions.hostname
+
+**Type**: `string`
+
+**Default**: `localhost`
+
+The hostname the dev server runs on.
+
+TODO: originally "The hostname where the browser tab will be open." is this more correct? Also why isn't it in the devoptions object in config.ts?
+
+### devOptions.port
+
+**Type**: `number`
+
+**Default**: `8080`
+
+The port the dev server runs on.
+
+### devOptions.fallback
+
+**Type**: `string`
+
+**Default**: `"index.html"`
+
+The HTML file to serve for non-resource routes.
+
+When using the Single-Page Application (SPA) pattern, this is the HTML "shell" file that gets served for every (non-resource) user route.
+
+⚠️ Make sure that you configure your production servers to serve this.
+
+### devOptions.bundle
+
+**Type**: `boolean`
+
+**Default**: TODO
+
+TODO: This is still in config.ts but seems deprecated
+
+### devOptions.open
+
+**Type**: `string`
+
+**Default**: `"**Default**"`
+
+Configures how the dev server opens in the browser when it starts.
+
+Any installed browser, e.g., "chrome", "firefox", "brave". Set "none" to disable.
+
+### devOptions.output
+
+**Type**: `"stream" | "dashboard"`
+
+**Default**: `"dashboard"`
+
+Set the output mode of the `dev` console:
+
+- `"dashboard"` delivers an organized layout of console output and the logs of any connected tools. This is recommended for most users and results in the best logging experience.
+- `"stream"` is useful when Snowpack is run in parallel with other commands, where clearing the shell would clear important output of other commands running in the same shell.
+
+### devOptions.hmr
+
+**Type**: `boolean`
+
+**Default**: `true`
+
+Toggles HMR on the Snowpack dev server.
+
+### devOptions.hmrDelay
+
+**Type**: `number` (milliseconds)
+
+**Default**: `0`
+
+Milliseconds to delay HMR-triggered browser update.
+
+### devoptions.hmrPort
+
+**Type**: `number`
+
+**Default**: `undefined`
+
+The port where Snowpack's HMR Websocket runs.
+
+The **Default** of `undefined` means it runs on the same port as the Snowpack dev server. TODO I think
+
+### devOptions.hmrErrorOverlay
+
+**Type**: `boolean`
+
+**Default**: `true`
+
+Toggles a browser overlay that displays JavaScript runtime errors when running HMR.
+
+### devOptions.out
+
+**Type**: `string`
+
+**Default**: `"build"`
+
+_NOTE:_ Deprecated, see `buildOptions.out`.
+
+## config.installOptions
+
+**Type**: `object` (option name: value)
+
+Configure how npm packages are installed.
+
+TODO: Had to pull from config.ts because these are in types.ts
+
+### installOptions.dest
+
+**Type**: `string`
+
+**Default**:`”web_modules”`
+
+Configures the destination for npm modules converted by Snowpack
```

Comment:
@FredKSchott also wrote this from scratch so could use a look (is it correct?)

Reply:
not actually an option that we want to expose to users. deleted, and removed from config